File: my-react-app/src/Models/Rating.ts
// Models/Rating.ts
export interface Rating {
  id: number;
  recipeId: number;
  userId: number;
  score: number; // e.g., rating out of 5
  comment?: string; // Optional comment
  createdAt: string; // ISO date string
  updatedAt: string; // ISO date string
}

File: Recipesdemo/src/main/java/Allrecipes/Recipesdemo/Entities/Rating.java
package Allrecipes.Recipesdemo.Entities;

import Allrecipes.Recipesdemo.Recipe.Recipe;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "ratings",
        uniqueConstraints = @UniqueConstraint(columnNames = {"recipe_id", "user_id"}))
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Rating {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name="recipe_id", nullable=false)
    private Recipe recipe;

    @ManyToOne
    @JoinColumn(name="user_id", nullable=false)
    private User user;

    @Column(nullable = false)
    private int score;

    @Column(length = 500)
    private String comment;


    @Column(nullable = false)
    private boolean deleted = false;
}
